src/core/mac/mac.cpp

```diff
@@ -770,13 +815,20 @@ void Mac::ReceiveDoneTask(Frame *aFrame, ThreadError aError)
         break;
 
     default:
+        error = kThreadError_Error;
         ExitNow();
     }
 
     // Source Whitelist Processing
     if (srcaddr.mLength != 0 && mWhitelist.IsEnabled())
     {
-        VerifyOrExit((entry = mWhitelist.Find(srcaddr.mExtAddress)) != NULL, ;);
+        entry = mWhitelist.Find(srcaddr.mExtAddress);
+
+        if (entry == NULL)
+        {
+            error = kThreadError_Error;
```

Comment:
You should be able to use `VerifyOrExit` with this.  Just add `error = kThreadError_Error` to the second argument.